builder/builder.go

```diff
@@ -1427,6 +1428,31 @@ func (b *Builder) NewDNFConfIfNeeded() error {
 	return nil
 }
 
+// getDirOwner returns the uid/gid owner of a path. There is no fully cross-
+// platform concept of "ownership", so this method only works on *nix systems.
+func getDirOwner(path string) (int, int, error) {
+	fi, err := os.Stat(path)
+	if err != nil {
+		return 0, 0, err
+	}
+	uid := int(fi.Sys().(*syscall.Stat_t).Uid)
+	gid := int(fi.Sys().(*syscall.Stat_t).Gid)
+	return uid, gid, nil
+}
+
+// getClosestExistingAncestor returns the longest substring of path that is an
+// existing path on the filesystem. If path itself already exists, path is
+// returned unchanged.
+func getClosestExistingAncestor(path string) (string, error) {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		path = filepath.Dir(path)
```

Comment:
Wouldn't it make more sense to have this outside the if statement? The name implies that you are getting an ancestor, so even if path exists, this should return its ancestor, not itself. Also, this would reduce one stat() call in the overall execution, since right now stat() is called twice on ServerStateDir

Reply:
@rchiossi good point. I wrote the function before I wrote how I was using it, and you're right -- as is there's no good reason for it to be in this order, 'cause it's only ever being called on cases where the full path does not exist.

Reply:
Heck, I can even just combine my two functions to reduce another stat, and just return the uid/gid of the closest parent. Maaaaybe someday someone will want to use either of these bits of functionality separately, but that need doesn't exist now.